### DISCUSSION TITLE
Import external components and kamelets

### DISCUSSION THREAD

**Discussion by rdobrik** [apache/camel-karavan#626] [Q&A] [answered]
First, I want to thank you for working on this amazing project! I am currently working on Apache Camel component for the Internet Computer blockchain platform (https://internetcomputer.org/) https://github.com/ic4j/ic4j-camel. (eventually also provide components for different smart contracts). I was also able to make it work with VS Code Karavan plug in, but it requires manual modification of component configuration files. (our project is for now outside of official Apache Camel repository, we are planning to move it there when it becomes more stable). Is there any way how to import it in VS Code? I understand you generate those files from Camel repository? Eventually we would like to have many custom Kamelets configured for specific smart contracts, or let our customers to do it for themself. Would be nice to have import option for components and kamelets.

Thanks,

Roman

**Comment by mgubaidullin** [comment, marked as answer]
Karavan already supports custom kamelets. For VS Code extension you could configure folder for custom kamelets in Settings

**Comment by durgapmenon** [comment]
Shifted the query from the Issues to Discussion.

One query.

Suppose, we create an Integration Route from Karavan and refactor the yaml file it into a custom kamelet based on a custom-kamelet example uploaded by Claus in GitHub.

Is it possible to bundle up this custom kamelet having a supporting Java Bean class and make it appear still in Karavan's Kamelet library?

I'm just curious to know if Karavan only picks up ***.kamelet.yaml files directly from the custom-kamelet folder configured in VS Code.

It would be a bit more structured if separate folders are present for each custom Kamelet also having custom business logic in the Java Bean class/component.

**Reply by davsclaus** [reply]
Its like you want to package a kamelet into a .jar where you have the yaml and .class and other supporting files.

In the kamelet you can refer to custom JARs by their maven corrdinates, that is what the OOTB kamelets does in the dependencies section. 

You can also embed code directly into the kamelet as .java / .groovy but that is likely only good for low amount of code, but can be okay when you need to define a bean, and have a bit of code to configure / tweak the bean etc.
